partner_multi_relation_parent/models/res_partner.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+
+    def update_relations(self, old_parent_id, parent_id=None):
+        if not parent_id:
+            parent_id = self.parent_id.id
+        par_rel_mod = self.env['res.partner.relation']
+        type_relation = self.env.ref(
+            'partner_multi_relation_parent.parent_relation_type'
+        ).id
+        # unlink previous relation
+        if old_parent_id:
+            previous = par_rel_mod.search(
+                [
+                    ('left_partner_id', '=', self.id),
+                    ('type_id', '=', type_relation),
+                    ('right_partner_id', '=', old_parent_id)
+                ]
+            )
+            previous.unlink()
+        # create new relations
+        par_rel_mod.create(
+            {'left_partner_id' : self.id,
+             'type_id': type_relation,
+             'right_partner_id': parent_id,
+            }
+        )
+
+    @api.model
+    def create(self, vals):
+        res = super(ResPartner, self).create(vals=vals)
+        if "parent_id" in vals.keys():
+            res.update_relations(None, vals['parent_id'])
+        return res
+
+    @api.multi
+    def write(self, vals , relation_create=True):
+        if relation_create:
+            for this in self:
+                if "parent_id" in vals.keys() and parent_id != False:
```

Comment:
again. do you use a lint checker locally?

Reply:
and we don't compare with a boolean in python

Reply:
#linter yes i have , and  it did not notify this poor form.
https://github.com/Therp/partner-contact/pull/2/commits/578ae9c4325db40bff7ba126b5dee563bffe27b7